terraform/charm/large_deployment/main.tf

```diff
@@ -0,0 +1,185 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+locals {
+  apps = [
+    for app in concat(var.apps != null ? var.apps : []) : app if app != null
+  ]
+
+  apps_in_main_model = [
+    for app in concat([var.failover], local.apps) :
+    app if app != null && app.model == var.main.model
+  ]
+  apps_not_in_main_model = [
```

Comment:
Same

Reply:
same rationale as above - i would like to avoid unnecessary offers, unless explicitly requested at deploy time with cross-models